packages/react-email/preview/src/pages/preview/[slug].tsx

```diff
@@ -56,15 +58,33 @@ const Preview: React.FC<Readonly<PreviewProps>> = ({
   reactMarkup,
   slug,
 }: any) => {
-  const [viewMode, setViewMode] = React.useState('desktop');
   const title = `${slug} — React Email`;
+  const router = useRouter();
+  const [viewMode, setViewMode] = React.useState('desktop');
+
+  function handleViewMode(mode: string) {
+    setViewMode(mode);
+    router.push({
+      pathname: router.pathname,
+      query: {
+        ...router.query,
+        view: `${mode}`,
```

Comment:
We can use only the variable here.
```suggestion
        view: mode,
```